plugins/woocommerce/src/StoreApi/Utilities/OrderController.php

```diff
@@ -171,6 +171,29 @@ public function validate_order_before_payment( \WC_Order $order ) {
 		$this->validate_email( $order );
 		$this->validate_selected_shipping_methods( $needs_shipping, $chosen_shipping_methods );
 		$this->validate_addresses( $order );
+
+
+    // Create a WP_Error object for custom validation
+    $validation_errors = new \WP_Error();
+
+    /**
+     * Allow plugins to perform custom validation before payment.
+     *
+     * @param \WC_Order $order             The order object.
+     * @param \WP_Error $validation_errors WP_Error object to add custom errors to.
+     */
+    do_action( 'woocommerce_checkout_validate_order_before_payment', $order, $validation_errors );
+
+    // Check if there are any errors after custom validation
+    if ( $validation_errors->has_errors() ) {
+        $error_messages = $validation_errors->get_error_messages();
+        $error_message = implode( ' ', $error_messages );
+        throw new RouteException(
+            'woocommerce_rest_checkout_custom_validation_error',
+            $error_message,
+            400
+        );
+    }
```

Comment:
Hi @ctrlaltdylan, thanks for your submission. Before this gets approved we will need indentation fixed as it won't pass [CI lints checks](https://github.com/woocommerce/woocommerce/actions/runs/10927833201/job/30344427808?pr=51513) and cannot be merged as is.

I would also suggest to consider usage of `apply_filter()` instead of `do_action()` which would be better suited to collect custom validation errors. Furthermore, to keep the consistency of `validate_order_before_payment()`, this could be encapsulated in it's own method, eg:

```suggestion

    // Perform custom validations
    $this->perform_custom_order_validation( $order );
}

/**
 * Perform custom order validation via WooCommerce hooks.
 *
 * Allows plugins to perform custom validation before payment.
 *
 * @param \WC_Order $order Order object.
 * @throws RouteException if validation fails.
 */
protected function perform_custom_order_validation( \WC_Order $order ) {
    // Create a WP_Error object for custom validation
    $validation_errors = apply_filters( 'woocommerce_checkout_validate_order_before_payment', new \WP_Error(), $order );

    // Check if there are any errors after custom validation
    if ( $validation_errors->has_errors() ) {
        throw new RouteException(
            'woocommerce_rest_checkout_custom_validation_error',
            implode( ' ', $validation_errors->get_error_messages() ),
            400
        );
    }
}
```

Reply:
Thanks for the suggestion @wavvves , I committed it

Reply:
I also replaced the spaces with tabs in a separate commit